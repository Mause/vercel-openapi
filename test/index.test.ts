import {expect, test} from "@oclif/test";
import {jestSnapshotPlugin} from "mocha-chai-jest-snapshot";
import chai from "chai";
import {parseDocument} from "yaml";

chai.use(jestSnapshotPlugin());

import cmd = require("../src");

describe("vercel-openapi", () => {
  test
    .stdout()
    .do(() => cmd.run(["generate", "test/fake"]))
    .it("runs", context => {
      expect(context.stdout).toMatchSnapshot();
    });

  test
    .stderr()
    .do(() => cmd.run(["generate", "test/fake", "--debug"]))
    .it("runs --debug", context => {
      expect(context.stderr).toMatchSnapshot();
    });

  test
    .stdout()
    .do(() => cmd.run(["generate", "test/fake", "--gitVersion"]))
    .it("runs --gitVersion", context => {
      expect(parseDocument(context.stdout).getIn(["info", "version"])).to.match(
        /^0.0.1\+[\da-z]{7}$/,
      );
    });
});

describe("scaffold", () => {
  test
    .stdout()
    .do(() =>
      cmd.run(["scaffold", "--title", "FakeAPI", "--version", "1.0.42"]),
    )
    .it("runs scaffold", context => {
      expect(context.stdout).toMatchSnapshot();
    });
});
