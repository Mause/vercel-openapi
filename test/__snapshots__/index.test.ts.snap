// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vercel-openapi runs hello 1`] = `
"{
  \\"openapi\\": \\"3.1.0\\",
  \\"info\\": {
    \\"title\\": \\"FakeAPI\\",
    \\"version\\": \\"0.0.1\\"
  },
  \\"paths\\": {
    \\"/api/create\\": {
      \\"post\\": {
        \\"operationId\\": \\"postCreate\\",
        \\"responses\\": {
          \\"default\\": {
            \\"$ref\\": \\"#/components/schemas/PostRequest\\"
          }
        },
        \\"requestBody\\": {
          \\"$ref\\": \\"#/components/schemas/PostRequest\\"
        }
      },
      \\"get\\": {
        \\"operationId\\": \\"getCreate\\",
        \\"responses\\": {
          \\"default\\": {
            \\"$ref\\": \\"#/components/schemas/PostRequest\\"
          }
        }
      }
    },
    \\"/api/endpoint\\": {
      \\"get\\": {
        \\"operationId\\": \\"getEndpoint\\",
        \\"responses\\": {
          \\"default\\": {
            \\"$ref\\": \\"#/components/schemas/Shape\\"
          }
        }
      }
    },
    \\"/api/dynamic/{dynamic_segment}\\": {
      \\"get\\": {
        \\"operationId\\": \\"getDynamic\\",
        \\"responses\\": {
          \\"default\\": {
            \\"$ref\\": \\"#/components/schemas/GetRequest\\"
          }
        }
      }
    }
  },
  \\"components\\": {
    \\"schemas\\": {
      \\"GetRequest\\": {
        \\"properties\\": {
          \\"name\\": {
            \\"type\\": \\"string\\"
          }
        },
        \\"type\\": \\"object\\",
        \\"required\\": [
          \\"name\\"
        ]
      },
      \\"PostRequest\\": {
        \\"properties\\": {
          \\"name\\": {
            \\"type\\": \\"string\\"
          }
        },
        \\"type\\": \\"object\\",
        \\"required\\": [
          \\"name\\"
        ]
      },
      \\"PostResponse\\": {
        \\"properties\\": {
          \\"id\\": {
            \\"type\\": \\"string\\"
          },
          \\"name\\": {
            \\"type\\": \\"string\\"
          }
        },
        \\"type\\": \\"object\\",
        \\"required\\": [
          \\"id\\",
          \\"name\\"
        ]
      },
      \\"Shape\\": {
        \\"properties\\": {
          \\"id\\": {
            \\"type\\": \\"string\\"
          }
        },
        \\"type\\": \\"object\\",
        \\"required\\": [
          \\"id\\"
        ]
      }
    },
    \\"securitySchemes\\": {}
  }
}
"
`;
